Unable to set oscillation on fan

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->

Running Hass.io on custom server

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->

ESP32 

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
Speed Fan

**Description of problem:**

I'm unable to set an oscillation output, the program will not build as it gives the following error:

> 'class esphome::speed::SpeedFan' has no member named 'set_oscillation'

This happens with a binary fan and a speed fan configuration
**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: ventilator
  platform: ESP32
  board: esp-wrover-kit
  includes:
    - fan.h

binary_sensor:
  - platform: gpio
    pin:
      number: 18
      mode: INPUT_PULLUP
    name: "SPEED"
    on_press:
      then:
      - fan.turn_on:
         id: ventilator
         speed: !lambda |-
          if (id(ventilator).speed == FAN_SPEED_LOW) {
            return FAN_SPEED_MEDIUM;
          } else if (id(ventilator).speed == FAN_SPEED_MEDIUM) {
            return FAN_SPEED_HIGH;
          } else {
            return FAN_SPEED_LOW;
          };
  - platform: gpio
    pin:
      number: 19
      mode: INPUT_PULLUP
    name: "ON/OFF"
    on_press:
      - fan.toggle: ventilator
fan:
 - platform: speed
   name: "Ventilator"
   id: ventilator
   output: speed_output
   oscillation_output: osc

output:
  - platform: custom
    type: float
    lambda: |-
      auto fan_output = new MyCustomFanoutput();
      App.register_component(fan_output);
      return {fan_output};
    outputs:
      id: speed_output
  
  - platform: gpio
    pin: 33
    id: osc
    inverted: true

    

```

**Logs (if applicable):**
<!--
Please copy the debug log here. If possible, also connect to the ESP over USB and copy those logs into the backticks.
-->
```
INFO Reading configuration...
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d /config/esphome/ventilator
Processing ventilator (platform: espressif32@1.6.0; framework: arduino; board: esp-wrover-kit)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp-wrover-kit.html
PLATFORM: Espressif 32 > Espressif ESP-WROVER-KIT
HARDWARE: ESP32 240MHz 320KB RAM (4MB Flash)
DEBUG: CURRENT(ftdi) ON-BOARD(ftdi) EXTERNAL(esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 27 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <AsyncTCP> 1.0.3
|-- <ESPmDNS> 1.0
|   |-- <WiFi> 1.0
|-- <Preferences> 1.0
|-- <WiFi> 1.0
|-- <Update> 1.0
Compiling /data/ventilator/.pioenvs/ventilator/src/main.cpp.o
Compiling /data/ventilator/.pioenvs/ventilator/lib090/WiFi/WiFiSTA.cpp.o
Compiling /data/ventilator/.pioenvs/ventilator/lib090/WiFi/WiFiScan.cpp.o
Compiling /data/ventilator/.pioenvs/ventilator/lib090/WiFi/WiFiServer.cpp.o
src/main.cpp: In function 'void setup()':
src/main.cpp:217:19: error: 'class esphome::speed::SpeedFan' has no member named 'set_oscillation'
speed_speedfan->set_oscillation(osc);
^
Compiling /data/ventilator/.pioenvs/ventilator/lib090/WiFi/WiFiUdp.cpp.o
Compiling /data/ventilator/.pioenvs/ventilator/lib4eb/ESPmDNS/ESPmDNS.cpp.o
*** [/data/ventilator/.pioenvs/ventilator/src/main.cpp.o] Error 1
========================== [ERROR] Took 4.17 seconds ==========================

```

**Additional information and things you've tried:**

I've also tried running the beta version of ESPHome to see if that resolves the issue, however this is not the case. The same error keeps popping up.


<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/esphome/components/binary/fan/__init__.py
+++ b/esphome/components/binary/fan/__init__.py
@@ -24,4 +24,4 @@ def to_code(config):
 
     if CONF_OSCILLATION_OUTPUT in config:
         oscillation_output = yield cg.get_variable(config[CONF_OSCILLATION_OUTPUT])
-        cg.add(var.set_oscillation(oscillation_output))
+        cg.add(var.set_oscillating(oscillation_output))

--- a/esphome/components/speed/fan/__init__.py
+++ b/esphome/components/speed/fan/__init__.py
@@ -29,4 +29,4 @@ def to_code(config):
 
     if CONF_OSCILLATION_OUTPUT in config:
         oscillation_output = yield cg.get_variable(config[CONF_OSCILLATION_OUTPUT])
-        cg.add(var.set_oscillation(oscillation_output))
+        cg.add(var.set_oscillating(oscillation_output))

